core/src/fnc/array.rs

```diff
@@ -133,6 +133,41 @@ pub fn distinct((array,): (Array,)) -> Result<Value, Error> {
 	Ok(array.uniq().into())
 }
 
+pub fn fill(
+	(mut array, value, start, end): (Array, Value, Option<isize>, Option<isize>),
+) -> Result<Value, Error> {
+	let min = 0;
+	let max = array.len();
+	let negative_max = -(max as isize);
+
+	let start = match start {
+		Some(start) if negative_max <= start && start < 0 => (start + max as isize) as usize,
+		Some(start) if start < negative_max => 0,
+		Some(start) => start as usize,
+		None => min,
+	};
+	let end = match end {
+		Some(end) if negative_max <= end && end < 0 => (end + max as isize) as usize,
+		Some(end) if end < negative_max => 0,
+		Some(end) => end as usize,
+		None => max,
+	};
+
+	if start == min && end >= max {
+		array.fill(value);
+	} else {
+		let range = start..end;
+
+		if let Some(subslice) = array.get_mut(range) {
+			for elem in subslice {
+				*elem = value.clone();
```

Comment:
Do you mean cloning from `start..(end - 1)`  and then moving (rather than cloning) the last one?

Something like this?

```Rust
        let end_minus_one = if end > 0 { end - 1 } else { 0 };

        for i in start..end_minus_one {
            if let Some(elem) = array.get_mut(i) {
                *elem = value.clone();
            }
        }

        if let Some(last_elem) = array.get_mut(end_minus_one) {
            *last_elem = value; // Move value into the last element without cloning
        }
```

Reply:
Well, I was hoping there was a way to avoid any cloning. But I suppose this is the best we can get. Thank you for sharing this.